s3/replication/manager/src/s3replicationmanager/manager_routes.py

```diff
@@ -76,22 +76,21 @@ async def add_job(request):
     """Handler to add job to job queue."""
     job_record = await request.json()
 
-    # Get first key and find if already present
-    job_id = job_record['Object-Name']
-    job = request.app['all_jobs'].get_job(job_id)
+    if job_record == {}:
+        return web.json_response('Empty json', status=500)
 
-    if job is not None:
-        return web.json_response(
-            {'ErrorResponse': 'job already present'}, status=201)
-    else:
-        job = request.app['all_jobs'].add_job_using_json(job_record)
-        LOG.debug('Added Job with job_id : {} '.format(job.get_job_id()))
-        LOG.debug(
-            'Added Job : {} '.format(
-                json.dumps(
-                    job,
-                    cls=JobJsonEncoder)))
-        return web.json_response({'job': job.get_dict()}, status=201)
+    job_prep = PrepareReplicationJob()
+
+    fdmi_job = job_prep.prepare_replication_job(job_record)
```

Comment:
names look reversed, fdmi_job here is actually converted replication job and job_record is fdmi_record

Reply:
done